src/solvers/blk_2m/update_rhs_blk_2m_common.hpp

```diff
@@ -118,7 +124,7 @@ void slvr_blk_2m_common<ct_params_t>::update_rhs(
   dot_rc = where(dot_rc * dt <= -rc, -rc / dt, dot_rc);
   dot_rr = where(dot_rr * dt <= -rr, -rr / dt, dot_rr);
   dot_nc = where(dot_nc * dt <= -nc, -nc / dt, dot_nc);
-  dot_nr = where(dot_nr * dt <= -rr, -rr / dt, dot_nr);
+  dot_nr = where(dot_nr * dt <= -nr, -nr / dt, dot_nr);
```

Comment:
@pdziekan I found this typo just now. Not sure if this might be causing problems? I will try to check tonight if I have time.

Reply:
From a short test, it seems like this wasn't causing the problems with spiky/small precip.

Reply:
#161 fixes this, thanks!